### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by hasanmumin** [komand/gosea#8]
Hello, 
Thank you for your article. it look really got. i just try to run your app, i got this error.
/hello works fine.
/users works fine (Unauthorized)
but the /tokens give me error *http2: panic serving [::1]:56586: runtime error: invalid memory address or nil pointer dereference* 
i follow your article step by step. And i just run your app.
Thanks.

**Comment by hasanmumin** [comment]
the problem 
```
	// Try to log in the user
	user, err := s.UserService.Read(u.ID)
```
line the UserService is not setting any reference it is nil.

**Comment by vgraveto** [comment]
Hello,

I had the same problem, it happens because **tokenService** _struct_ has the field **UserService** as an _interface_ of type **UserService** and it should be an implementation of that _interface_ the **userService**, regard on the small caps of the user.

 The solution is on the declaration of tokenService struct to be:
type tokenService struct {
	UserService userService
}

This article is very good thanks. Where can I find part 2 @rossnanop .
Vitor

**Comment by rossnanop** [comment]
@hasanmumin @vgraveto fixed in #6 

P.S. I'll try to scrounge up some time to write a new piece and update this repo with some new code.
